examples/siro_sandbox/sandbox_app.py

```diff
@@ -460,38 +477,44 @@ def sim_update(self, dt):
                 self.cam_zoom_dist /= (
                     1.0 + self.gui_input.mouse_scroll_offset * zoom_sensitivity
                 )
-            max_zoom_dist = 50.0
-            min_zoom_dist = 0.1
             self.cam_zoom_dist = mn.math.clamp(
-                self.cam_zoom_dist, min_zoom_dist, max_zoom_dist
+                self.cam_zoom_dist, self._min_zoom_dist, self._max_zoom_dist
             )
 
+        # two ways for camera pitch and yaw control for UX comparison:
+        # 1) hold WASD keys
+        self._camera_pitch_and_yaw_wasd_control()
+        # 2) hold R and move mouse
+        self._camera_pitch_and_yaw_mouse_control()
+
         agent_idx = self.ctrl_helper.get_gui_controlled_agent_index()
-        if agent_idx is not None:
+        if agent_idx is None:
+            self._free_camera_lookat_control()
+            lookat = self.lookat
+        else:
             art_obj = (
                 self.get_sim().agents_mgr[agent_idx].articulated_agent.sim_obj
             )
             robot_root = art_obj.transformation
             lookat = robot_root.translation + mn.Vector3(0, 1, 0)
-        else:
-            self._free_camera_lookat_control()
-            lookat = self.lookat
 
-        # two ways for camera pitch and yaw control for UX comparison:
-        # 1) hold WASD keys
-        self._camera_pitch_and_yaw_wasd_control()
-        # 2) hold R and move mouse
-        self._camera_pitch_and_yaw_mouse_control()
+        if self._first_person_mode:
+            self.cam_zoom_dist = self._min_zoom_dist
+            lookat += 0.075 * robot_root.backward
+            lookat -= mn.Vector3(0, 0.2, 0)
+
         offset = mn.Vector3(
             np.cos(self.lookat_offset_yaw) * np.cos(self.lookat_offset_pitch),
             np.sin(self.lookat_offset_pitch),
             np.sin(self.lookat_offset_yaw) * np.cos(self.lookat_offset_pitch),
         )
+
         cam_transform = mn.Matrix4.look_at(
             lookat + offset.normalized() * self.cam_zoom_dist,
```

Comment:
Note my comments here: https://cvmlp.slack.com/archives/D03BA68KGHW/p1683043849141029

An FPS camera should have a fixed position, unlike here, where the position changes as the offset changes.

Reply:
Yes, remember that. Currently, this is a quick hack: if zoom dist is set to a small fixed value this is almost equivalent to rotating fixed camera at the spot. This allows to reuse a lot of code without duplication.

Otherwise, I have to create two separate look-at matrices: one for 3-rd person view where the camera is rotating around the agent and the other one for FPS where the camera is fixed. And all the yaw/pitch controls will have an opposite effect on each camera (increasing yaw/pitch angle will make agent look up/left in on case and down/right in the other), right?

Reply:
I agree with your assessment. It would be cleaner to eventually have the FPS camera math implemented correctly, instead of your quick hack. But this is low priority.